sympy/series/tests/test_order.py

```diff
@@ -88,7 +88,9 @@ def test_simple_8():
     assert O(x**2*sqrt(x)) == O(x**(S(5)/2))
     assert O(x**3*sqrt(-(-x)**3)) == O(x**(S(9)/2))
     assert O(x**(S(3)/2)*sqrt((-x)**3)) == O(x**3)
-    assert O(x*(-2*x)**(I/2)) == O(x*(-x)**(I/2))
+    # not sure why the first doesn't become the 3rd without
+    # iteration in the Order.__new__ routine.
+    assert O(x*(-2*x)**(I/2)) == O(x*(-x)**(I/2)) == O((-x)**(1 + I/2))
```

Comment:
It seems that the order algorithm should be improved. By definition, `(-2*x)**(I/2)` is `exp(log(-2*x)*I/2)`. Assuming that the the principal branch of the logarithm should be used, the imaginary part of `log(-2*x)` is between `-pi` and `pi`. Hence the real part of `log(-2*x)*I/2` is between `-pi/2` and `pi/2`, and therefore the absolute value of `(-2*x)**(I/2)` is in the bounded interval `[exp(-pi/2), exp(pi/2)]`. This means the the order of `x*(-2*x)**(I/2)` is `O(x)`.

Reply:
I think it is ok to commit this. Fixing Order is another matter and independent of powsimp. (It looks like imaginary exponents should have no effect on order.)